instrumentation/akka/akka-http-10.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/akkahttp/client/HttpExtClientInstrumentation.java

```diff
@@ -43,55 +46,81 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class SingleRequestAdvice {
 
+    public static class AdviceScope {
+      private final Context context;
+      private final Scope scope;
+      private final HttpRequest request;
+
+      private AdviceScope(Context context, Scope scope, HttpRequest request) {
+        this.context = context;
+        this.scope = scope;
+        this.request = request;
+      }
+
+      public static Object[] start(HttpRequest request) {
+        Context parentContext = currentContext();
+        if (!instrumenter().shouldStart(parentContext, request)) {
+          return new Object[] {null, request};
+        }
+        Context context = instrumenter().start(parentContext, request);
+        // Making context current is required for header context propagation to work as expected
+        // because it implicitly relies on the current context when injecting headers.
+        Scope scope = context.makeCurrent();
+        // Request is immutable, so we have to assign new value once we update headers
+        HttpRequest modifiedRequest = setter().inject(request);
+        AdviceScope adviceScope = new AdviceScope(context, scope, modifiedRequest);
+        return new Object[] {adviceScope, modifiedRequest};
+      }
+
+      public Future<HttpResponse> end(
+          @Nullable ActorSystem actorSystem,
+          @Nullable Future<HttpResponse> responseFuture,
+          @Nullable Throwable throwable) {
+
+        scope.close();
+        if (actorSystem == null) {
+          return responseFuture;
+        }
+        if (throwable == null) {
+          responseFuture.onComplete(
+              new OnCompleteHandler(context, request), actorSystem.dispatcher());
+          responseFuture =
+              FutureWrapper.wrap(responseFuture, actorSystem.dispatcher(), currentContext());
+        } else {
+          instrumenter().end(context, request, null, throwable);
+        }
+        return responseFuture;
+      }
```

Comment:
Nit: avoiding assignment to parameter
```suggestion
        if (actorSystem != null) {
          if (throwable == null) {
            responseFuture.onComplete(
                new OnCompleteHandler(context, request), actorSystem.dispatcher());
            return FutureWrapper.wrap(responseFuture, actorSystem.dispatcher(), currentContext());
          } else {
            instrumenter().end(context, request, null, throwable);
          }
        }
        return responseFuture;
      }
```